### PR TITLE
drive-by cleanups

### PR DESCRIPTION
**Pull request by emberian**
main changes:
- added rust/agents, moved agents there
- moved rust/optics-ethereum to rust/chains
- ran shellcheck on shell scripts, fixed some nits

### CODE REVIEW

**Comment by emberian** [comment]
TODO: sign a commit at the tail. @prestwich, should i re-sign all the interim commits as well?

**Comment by ErinHales** [comment]
> TODO: sign a commit at the tail. @prestwich, should i re-sign all the interim commits as well?

Pretty sure they all need to be signed. This is the easiest way to do it, then you should set up auto signing.

**Comment by prestwich** [comment]
apologies for pushing to main a few times and causing conflicts. have been trying to fix an issue with deployed agents

I can rebase and sign everything if you'd like. otherwise, turning on signing by default and rebasing to latest main will sign all commits

**Comment by prestwich** [comment]
for this to be merged:
1. rebase onto main and resolve conflicts
2. sign commits
3. check with @yourbuddyconner to ensure that it doesn't break any docker/helm work

File: solidity/optics-core/update_abis.sh [hunk 1]
```diff
@@ -1,3 +1,7 @@
-cat artifacts/contracts/Replica.sol/Replica.json| jq .abi > ../../rust/optics-ethereum/abis/Replica.abi.json && \
-cat artifacts/contracts/Home.sol/Home.json| jq .abi > ../../rust/optics-ethereum/abis/Home.abi.json && \
-cat artifacts/contracts/XAppConnectionManager.sol/XAppConnectionManager.json | jq .abi > ../../rust/optics-ethereum/abis/XAppConnectionManager.abi.json
\ No newline at end of file
+#!/bin/sh 
+
+copy() {
+    jq .abi < artifacts/contracts/"$1".sol/$.json > ../rust/chains/optics-ethereum/abis/"$1".abi.json
+}
```
Comment (prestwich) [inline solidity/optics-core/update_abis.sh:5]:
```
copy() {
    jq .abi < artifacts/contracts/"$1".sol/"$1".json > ../../rust/chains/optics-ethereum/abis/"$1".abi.json
}
```

File: rust/optics-core/src/lib.rs [hunk 2]
```diff
@@ -145,15 +145,258 @@ impl Signer for Signers {
...
 
-#[async_trait]
-trait SignerExt: Signer {
-    async fn sign_message_without_eip_155<S: Send + Sync + AsRef<[u8]>>(
-        &self,
-        message: S,
-    ) -> Result<Signature, <Self as Signer>::Error> {
-        let mut signature = self.sign_message(message).await?;
-        signature.v = 28 - (signature.v % 2);
-        Ok(signature)
```
Comment (prestwich) [inline rust/optics-core/src/lib.rs:156]: Looks like a messy rebase mostly resotred an old version of this file.

Here are what you need to do to fix:

- Structs in `lib` ought to deleted as they've been moved to `types/`
- `SignerExt` needs to be restored

**Comment by prestwich** [comment]
tests fail because `processor` and `optics-core` use a shared test vectors file in `vectors/`. They open it with a utility function in `optics-core` which has the relative path hardcoded. Now that path is different for each crate, as processor has moved

**Comment by emberian** [comment]
> tests fail because `processor` and `optics-core` use a shared test vectors file in `vectors/`. They open it with a utility function in `optics-core` which has the relative path hardcoded. Now that path is different for each crate, as processor has moved

what do you think about leaning on cargo for these paths? from the docs:

```
CARGO_TARGET_TMPDIR — Only set when building integration test or benchmark code. This is a path to a directory inside the target directory where integration tests or benchmarks are free to put any data needed by the tests/benches. Cargo initially creates this directory but doesn't manage its content in any way, this is the responsibility of the test code. There are separate directories for debug and release profiles.
```

relative paths drive me insane, i have such a hard time keeping track of cwd in my head.

File: rust/optics-core/src/test_utils.rs [hunk 3]
```diff
@@ -16,9 +16,21 @@ pub struct MerkleTestCase {
     pub expected_root: H256,
 }
 
+/// Find a vector file assuming that a git checkout exists
+// TODO: look instead for the workspace `Cargo.toml`? use a cargo env var?
+pub fn find_vector(final_component: &str) -> PathBuf {
```
Comment (emberian) [inline rust/optics-core/src/test_utils.rs:21]: i ended up doing this instead of anything with a cargo env var.

File: scripts/test-all.sh [hunk 4]
```diff
@@ -6,6 +6,8 @@ set -xe
 
 # update ABIs
 (
+    cd ./solidity
```
Comment (emberian) [inline scripts/test-all.sh:9]: this was removed by accident during rebase

File: rust/optics-core/src/test_utils.rs (hunk 3 above)
Comment (prestwich) [reply rust/optics-core/src/test_utils.rs:21]: cool. this looks fine. would be good to add a `.take(4)` to limit how deep into the directory structure we can go

**Comment by prestwich** [comment]
relative paths also drive me insane. at some point we should do a cleanup and get rid of all of them

**Comment by prestwich** [comment]
give that checks have all passed, I'll merge this once we get approves from 
- @yourbuddyconner (please review the dockerfile changes and ensure this doesn't affect Helm work)
- @ErinHales (please review the .sh script changes)

File: rust/chains/README.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+To add a new chain:
+
+    - add a new crate here
```
Comment (ErinHales) [inline rust/chains/README.md:3]: Assuming "here" is supposed to have a link?

**Review by yourbuddyconner — APPROVED** [review]
Dockerfile changes ok

File: rust/chains/README.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
 To add a new chain:
 
-    - add a new crate here
-    - add a new enum variant [there](optics_base::homes::Homes) # TODO: rich text link from md to live rust
+    - add a new crate to this directory, implementing the appropriate traits.
```
Comment (emberian) [inline rust/chains/README.md:3]: @ErinHales this is much less ambiguous!

**Comment by ErinHales** [comment]
Getting these errors when running `cargo clippy`
Screen Shot 2021-08-20 at 3 18 34 PM

File: pre-commit.sh [hunk 7]
```diff
@@ -153,9 +153,10 @@ else
 fi
 
 # Git add abis if updated
-if ! git diff-index --quiet HEAD -- ./rust/optics-ethereum/abis/; then
-    echo '+git add ./rust/optics-ethereum/abis/*'
-    git add ./rust/optics-ethereum/abis/*
+abis_dir=./rust/chains/optics-ethereum/abis/
+if ! git diff-index --quiet HEAD -- ./rust/chains/optics-ethereum/abis/; then
```
Comment (ErinHales) [inline pre-commit.sh:157]: Use `abis_dir` in conditional: `if ! git diff-index --quiet HEAD -- abis_dir; then`

**Comment by emberian** [comment]
> Getting these errors when running `cargo clippy`

@ErinHales i also got those, doing a `rustup update` fixed it for me -- i'm currently using `rust 1.54`. i'm going to commit that as a version pin...

**Review by ErinHales — APPROVED** [review]

**Review by prestwich — APPROVED** [review]

**Pull request merged**